src/aihwkit/utils/analog_info.py

```diff
@@ -0,0 +1,203 @@
+from aihwkit.nn import AnalogLinear
+
+COLUMN_NAMES = {
+    "name": "Layer Name",
+    "isanalog": "Analog/Digital",
+    "input_size": "Input Shape",
+    "output_size": "Output Shape",
+    "kernel_size": "Kernel Shape",
+    "num_tiles": "Number of Tiles",
+    "macs": "Multi_Adds",
+    "reuse_factor": "Reuse Factor"
+}
+
+INPUT_SIZE_TYPE = Any
+FORMATTING_WIDTH = 150
+
+
+class LayerInfo:
+    """
+    class storing layer statistics and information.
+    """
+    # pylint: disable=too-many-instance-attributes
+    module: nn.Module
+    name: str
+    isanalog: bool
+    num_tiles: int
+    input_size: INPUT_SIZE_TYPE
+    output_size: INPUT_SIZE_TYPE
+    kernel_size: INPUT_SIZE_TYPE
+    reuse_factor: int
+
+    def __init__(self, module: nn.Module, rpu_config: Optional[RPUConfigAlias] = None):
+        self.module = module
+        self.name = self.module.__class__.__name__
+        self.isanalog = isinstance(self.module, AnalogModuleBase)
+        self.num_tiles = 0 if not self.isanalog else self.module._analog_tile_counter
+        self.kernel_size = None
+        self.reuse_factor = 0
+        self.input_size, self.output_size = None, None
+        self.rpu_config = rpu_config
+
+    def __set_layer_size(self,
+                         input_size: Optional[INPUT_SIZE_TYPE],
+                         output_size: Optional[INPUT_SIZE_TYPE]) -> None:
+        """Private method to set layer's sizes attributes."""
+        self.input_size = input_size
+        self.output_size = output_size
+
+    def __set_reuse_factor(self, reuse_factor: int) -> None:
+        """Private method to set layer's reuse factor attribute."""
+        self.reuse_factor = reuse_factor
+
+    def calculate_size(self,
+                       input_size: Optional[INPUT_SIZE_TYPE]
+                       ) -> Tuple[INPUT_SIZE_TYPE, INPUT_SIZE_TYPE]:
+        """Set input_size or output_size of the layer."""
+        rand_tensor = torch.rand(input_size)
+        output_size = self.module(rand_tensor).shape
+        self.__set_layer_size(input_size, tuple(output_size))
+        return input_size, tuple(output_size)
+
+    def set_kernel_size(self) -> None:
+        """Set kernel size attribute."""
+        if hasattr(self.module, 'kernel_size'):
+            self.kernel_size = self.module.kernel_size
+
+    def calculate_reuse_factor(self) -> None:
+        """Compute the reuse factor.
+
+        The reuse factor is the number of vector matrix multiplication a layer computes."""
+        if isinstance(self.module, AnalogLinear):
+            ruf = reduce(operator.mul, (self.input_size), 1) // int(self.input_size[-1])
+            self.__set_reuse_factor(ruf)
+        elif isinstance(self.module, _AnalogConvNd):
+            # TODO: Extend the formula to ConvNd
+            ruf = reduce(operator.mul, (self.output_size), 1) // self.output_size[1]
+            self.__set_reuse_factor(ruf)
+
+    def layer_summary_dict(self) -> dict:
```

Comment:
Can one add the (logical) tile size, ie the weight matrix size stored on the tile?  Also utilization might be good to add later, once we map to multiple tiles (will be possible soon)